tests/subsys/settings/functional/src/settings_basic_test.c

```diff
@@ -657,3 +690,63 @@ ZTEST(settings_functional, test_direct_loading_filter)
 	}
 	settings_deregister(&filtered_loader_settings);
 }
+
+#define TEST_VALUE 0xfeedda7a
+ZTEST(settings_functional, test_persistence)
+{
+	// TODO: Z_TEST_SKIP_IFNDEF(CONFIG_TEST_SETTINGS_PERSISTENCE);
```

Comment:
Here's a WIP test. It should probably have a separate test suite, but not sure. It is important for ITS and PS, since they write to storage using a delayed work, and are basically RAM-backed in runtime. So to make sure the PSA side of things actually works we need to reboot and validate it persists.

Could be included in the samples/subsys/settings as well I think, since that sample could showcase that after a reboot your settings are still there. It will also be used in the integration test, just as if in `tests/`